homeassistant/components/sensor/juhe_stock.py

```diff
@@ -0,0 +1,165 @@
+    vol.Required(CONF_KEY): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the Aliyun_stock sensor."""
+
+    symbols = config.get(CONF_SYMBOLS)
+    key = config.get(CONF_KEY)
+
+    dev = []
+    for symbol in symbols:
+        data = JuheStockData(hass, symbol, key)
+        dev.append(JuheStockSensor(data, symbol))
+
+    async_add_devices(dev, True)
+
+
+class JuheStockSensor(Entity):
+    """Representation of a Juhe Stock sensor."""
+
+    def __init__(self, data, symbol):
+        """Initialize the sensor."""
+        self.data = data
+        self._symbol = symbol
+        self._state = None
+        self._unit_of_measurement = '元'
+        self._name = symbol
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._state is not None:
+            return {
+                ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+                ATTR_OPEN: self.data.price_open,
+                ATTR_PREV_CLOSE: self.data.prev_close,
+                ATTR_HIGH: self.data.high,
+                ATTR_LOW: self.data.low,
+                ATTR_NAME: self.data.name,
+            }
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return ICON
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Get the latest data and updates the states."""
+        _LOGGER.debug("Updating sensor %s - %s", self._name, self._state)
+        self.data.update()
+        self._state = self.data.state
+
+
+class JuheStockData(object):
+    """Get data from Juhe stock imformation."""
+
+    def __init__(self, hass, symbol, key):
+        """Initialize the data object."""
+
+        self._symbol = symbol
+        self.state = None
+        self.price_open = None
+        self.prev_close = None
+        self.high = None
+        self.low = None
+        self.name = None
+        self.hass = hass
+
+        self.host = 'web.juhe.cn:8080'
+        self.url = "/finance/stock/hs?gid=" + self._symbol + "&key=" + key
+
+    def update(self):
+        """Get the latest data and updates the states."""
+        conn = http.client.HTTPConnection(self.host)
+        conn.request("GET", self.url)
+        result = conn.getresponse()
+
+        if (result.status != 200):
+            _LOGGER.error("Error http reponse: %d", result.status)
+            return
+
+        # data = eval(result.read())
+        data = json.loads(str(result.read(), encoding='utf-8'))

```

Comment:
Isn't is possible to use `result.json()`?

Reply:
Thank you so much for your detailed feedback! This is actually my first time participating in open-source development, so my code may seem a little clumsy. I will definitely change my code according to your suggestions.